File: app/application/fizz_buzz_value_command.go
package application

import "fizzbuzz/domain/type"

// FizzBuzzValueCommand 値オブジェクトを返すコマンド
type FizzBuzzValueCommand struct {
	fizzBuzzType types.FizzBuzzType
}

// NewFizzBuzzValueCommand FizzBuzzValueCommandのコンストラクタ
func NewFizzBuzzValueCommand(fizzBuzzType types.FizzBuzzType) *FizzBuzzValueCommand {
	return &FizzBuzzValueCommand{fizzBuzzType: fizzBuzzType}
}

// Execute 値オブジェクトを生成して値を返す
func (cmd *FizzBuzzValueCommand) Execute(number int) string {
	value := cmd.fizzBuzzType.Generate(number)
	return value.Value()
}

File: app/src/Application/FizzBuzzValueCommand.php
<?php

namespace App\Application;

use App\Domain\Type\FizzBuzzType;

class FizzBuzzValueCommand implements FizzBuzzCommand
{
    private FizzBuzzType $type;

    public function __construct(FizzBuzzType $type)
    {
        $this->type = $type;
    }

    public function execute(int $number): string
    {
        $value = $this->type->generate($number);
        return $value->getValue();
    }
}
